retrofit/src/test/java/retrofit/RetrofitTest.java

```diff
@@ -291,12 +295,64 @@
         .addConverterFactory(new MyConverterFactory())
         .build();
     Annotated annotated = retrofit.create(Annotated.class);
-    annotated.parameter(null); // Trigger internal setup.
+    annotated.bodyParameter(null); // Trigger internal setup.
 
     Annotation[] annotations = annotationsRef.get();
     assertThat(annotations).hasAtLeastOneElementOfType(Annotated.Foo.class);
   }
 
+  @Test public void parameterAnnotationsPassedToStringConverter() {
+    final AtomicReference<Annotation[]> annotationsRef = new AtomicReference<>();
+    class MyConverterFactory extends Converter.Factory {
+      @Override public Converter<?, String> toString(Type type, Annotation[] annotations) {
+        annotationsRef.set(annotations);
+
+        return new Converter<Object, String>() {
+          @Override public String convert(Object value) throws IOException {
+            return String.valueOf(value);
+          }
+        };
+      }
+    }
+    Retrofit retrofit = new Retrofit.Builder()
+        .baseUrl(server.url("/"))
+        .addConverterFactory(new MyConverterFactory())
+        .build();
+    Annotated annotated = retrofit.create(Annotated.class);
+    annotated.queryParameter(null); // Trigger internal setup.
+
+    Annotation[] annotations = annotationsRef.get();
+    assertThat(annotations).hasAtLeastOneElementOfType(Annotated.Foo.class);
+  }
+
+  @Test public void stringConverterNotCalledForString() {
+    class MyConverterFactory extends Converter.Factory {
+      @Override public Converter<?, String> toString(Type type, Annotation[] annotations) {
+        throw new AssertionError();
+      }
+    }
+    Retrofit retrofit = new Retrofit.Builder()
+        .baseUrl(server.url("/"))
+        .addConverterFactory(new MyConverterFactory())
+        .build();
+    CallMethod service = retrofit.create(CallMethod.class);
+    service.queryString(null);
+  }
+
+  @Test public void stringConverterReturningNullResultsInDefault() {
+    class MyConverterFactory extends Converter.Factory {
+      @Override public Converter<?, String> toString(Type type, Annotation[] annotations) {
+        return null;
+      }
+    }
+    Retrofit retrofit = new Retrofit.Builder()
+        .baseUrl(server.url("/"))
+        .addConverterFactory(new MyConverterFactory())
+        .build();
+    CallMethod service = retrofit.create(CallMethod.class);
+    service.queryObject(null);
```

Comment:
Yeah. I'll assert the `Call` is non-null and that the supplied factory was called.